eZ/Bundle/EzPublishCoreBundle/FieldType/RichText/Converter/Html5Input.php

```diff
@@ -22,4 +23,27 @@ public function __construct($stylesheet, ConfigResolverInterface $configResolver
         $customStylesheets = $customStylesheets ?: array();
         parent::__construct($stylesheet, $customStylesheets);
     }
+
+    /**
+     * Method checks if document contains any links with '[' or ']' sign an replaces it to its html entities version.
+     *
+     * @param DOMDocument $document
+     *
+     * @return DOMDocument
+     */
+    public function convert(DOMDocument $document)
+    {
+        // EZP-27562 - links containing [ or ] sign results in xml validation error
+        foreach ($document->getElementsByTagName('a') as $link) {
+            $href = $link->getAttribute('href');
+            if ($href !== '') {
+                $href = str_replace('[', '%5B', $href);
+                $href = str_replace(']', '%5D', $href);
```

Comment:
we had a discussion today with @andrerom and idea popped up to remove anyURI clause from [dockbook.rng](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/FieldType/RichText/Resources/schemas/docbook/docbook.rng#L206) as apparently it does not work correctly. In that case all of code in this PR will be irrelevant as validation will be successful. @pspanja what do you think about such idea? (I ask you since you are one who committed `dockbook.rng`)